analyzer/windows/modules/auxiliary/autoruns.py

```diff
@@ -0,0 +1,145 @@
+import os
+import csv
+import logging
+import subprocess
+import platform
+import shlex
+from winreg import SetValueEx, CreateKeyEx, REG_DWORD, HKEY_CURRENT_USER, KEY_ALL_ACCESS
+
+from lib.common.abstracts import Auxiliary
+from lib.core.config import Config
+from lib.common.exceptions import CuckooPackageError
+from lib.common.results import upload_to_host
+
+log = logging.getLogger(__name__)
+
+
+__author__ = "[Canadian Centre for Cyber Security] @CybercentreCanada"
+
+
+class Autoruns(Auxiliary):
+    """Autoruns from sysinternals"""
+
+    def __init__(self, options, config):
+        Auxiliary.__init__(self, options, config)
+        self.config = Config(cfg="analysis.conf")
+        self.enabled = self.config.autoruns
+        self.output_dir = "C:\\\\autoruns"
+        self.output_file_start = "autoruns_start.txt"
+        self.output_file_end = "autoruns_end.txt"
+        self.output_file_diff = "autoruns.diff"
+        self.startupinfo = subprocess.STARTUPINFO()
+        self.startupinfo.dwFlags |= subprocess.STARTF_USESHOWWINDOW
+
+        if not os.path.exists(self.output_dir):
+            os.makedirs(self.output_dir)
+            #reg.exe ADD "HKCU\Software\Sysinternals\Autoruns" /v EulaAccepted /t REG_DWORD /d 1 /f
+        try:
+            with CreateKeyEx(HKEY_CURRENT_USER, "Software\Sysinternals\Autoruns", 0, KEY_ALL_ACCESS) as key:
+                SetValueEx(key, "EulaAccepted", 0, REG_DWORD, 1)
+        except OSError as e:
+            pass
+
+    def start(self):
+        # First figure out what architecture the system in running (64 or 86)
+        bin_path = os.path.join(os.getcwd(), "bin")
+
+        if "AMD64" in platform.uname():
```

Comment:
you can set the platform in a class variable here so that you don't need to check the platform in `stop()`